client/Lykke.AlgoStore.MatchingEngineAdapter.Client/MatchingEngineAdapterClient.cs

```diff
@@ -0,0 +1,77 @@
+﻿using Common.Log;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain;
+using Lykke.AlgoStore.MatchingEngineAdapter.Abstractions.Domain.Listening.Requests;
+using System;
+using System.Threading.Tasks;
+
+namespace Lykke.AlgoStore.MatchingEngineAdapter.Client
+{
+    internal class MatchingEngineAdapterClient : IMatchingEngineAdapterClient
+    {
+        private readonly IRequestManager _requestManager;
+        private readonly ILog _log;
+
+        public MatchingEngineAdapterClient(IRequestManager requestManager, ILog log)
+        {
+            _requestManager = requestManager ?? throw new ArgumentNullException(nameof(requestManager));
+            _log = log ?? throw new ArgumentNullException(nameof(log));
+        }
+
+        public void SetClientAndInstanceId(string clientId, string instanceId)
+        {
+            _requestManager.SetClientAndInstanceId(clientId, instanceId);
+        }
+
+        public Task<string> Ping(string content)
+        {
+            return Task.Run(() => PingSync(content));
+        }
+
+        private string PingSync(string content)
+        {
+            var pingRequest = new PingRequest { Message = content };
+
+            _log.WriteInfo(nameof(MatchingEngineAdapterClient), nameof(PingSync),
+                           $"Sending MEA Ping request with content {content}");
+
+            (var waitHandle, var requestId) = _requestManager.MakeRequest(MeaRequestType.Ping, pingRequest);
+
+            waitHandle.WaitOne();
```

Comment:
you can use async locks instead of blocking thread

var semaphoreSlim = new SemaphoreSlim(1, 1);
await semaphoreSlim.WaitAsync();